dlme_airflow/services/harvest_dag_generator.py

```diff
@@ -40,52 +44,67 @@ def default_dag_args():
     }
 
 
-def create_dag(provider, default_args) -> DAG:
+def create_dags(provider) -> list[DAG]:
+    """Returns a list of DAGs for a given provider."""
+    dags = []
+    if provider.catalog.metadata.get("separate_dags"):
+        # create a DAG for each collection, which can be useful if you want to
+        # schedule them separately
+        for collection in provider.collections:
+            dags.append(assemble_dag(collection))
+    else:
+        # create a single DAG for the provider and all its collections which
+        # will all run in parallel
+        dags.append(assemble_dag(provider))
+    return dags
+
+
+def assemble_dag(source: Union[Provider, Collection]):
```

Comment:
`assemble_dag()` is responsible for setting up a DAG either for a Provider or an individual Collection. It will call either `build_provider_etl_taskgroup()` or `build_collection_etl_taskgroup()` depending on what is passed in.